frontend/src/component/common/PermissionButton/PermissionButton.tsx

```diff
@@ -19,85 +22,110 @@ export interface IPermissionButtonProps extends Omit<ButtonProps, 'title'> {
-                                <ConditionallyRender
-                                    condition={!access}
-                                    show={<Lock titleAccess="Locked" />}
-                                    elseShow={
-                                        Boolean(rest.endIcon) && rest.endIcon
-                                    }
-                                />
-                            </>
-                        }
-                    >
-                        {children}
-                    </Button>
-                </span>
-            </TooltipResolver>
+const RootPermissionButton: React.FC<IPermissionButtonProps> = React.forwardRef(
+    (props, ref) => {
+        const access = useHasRootAccess(
+            props.permission,
+            props.environmentId,
+            props.projectId
         );
+
+        return <BasePermissionButton {...props} access={access} ref={ref} />;
+    }
+);
+
+const BasePermissionButton: React.FC<IPermissionBaseButtonProps> =
+    React.forwardRef(
+        (
+            {
+                permission,
+                access,
+                variant = 'contained',
+                color = 'primary',
+                onClick,
+                children,
+                disabled,
+                projectId,
+                environmentId,
+                tooltipProps,
+                ...rest
+            },
+            ref
+        ) => {
+            const id = useId();
+
+            return (
+                <TooltipResolver
+                    {...tooltipProps}
+                    title={formatAccessText(access, tooltipProps?.title)}
+                    arrow
+                >
+                    <span id={id}>
+                        <Button
+                            ref={ref}
+                            onClick={onClick}
+                            disabled={disabled || !access}
+                            aria-labelledby={id}
+                            variant={variant}
+                            color={color}
+                            {...rest}
+                            endIcon={
+                                <>
+                                    <ConditionallyRender
+                                        condition={!access}
+                                        show={<Lock titleAccess="Locked" />}
+                                        elseShow={
+                                            Boolean(rest.endIcon) &&
+                                            rest.endIcon
+                                        }
+                                    />
+                                </>
+                            }
+                        >
+                            {children}
+                        </Button>
+                    </span>
+                </TooltipResolver>
+            );
+        }
+    );
+
+const PermissionButton: React.FC<IPermissionButtonProps> = React.forwardRef(
+    (props, ref) => {
```

Comment:
same pattern in 3 components. Check if projectId and environmentId are provided. If the are pass props and ref to component doing root access and change request check. Otherwise go to the component doing only root access check.